src/partition/repart.c

```diff
@@ -2232,40 +2256,42 @@ static int context_acquire_partition_uuids_and_labels(Context *context) {
         assert(context);
 
         LIST_FOREACH(partitions, p, context->partitions) {
-                assert(sd_id128_is_null(p->new_uuid));
+                if (sd_id128_is_null(p->new_uuid)) {
+                        /* Not explicitly set by user! */
 
-                /* Never touch foreign partitions */
-                if (PARTITION_IS_FOREIGN(p)) {
-                        p->new_uuid = p->current_uuid;
+                        /* Never touch foreign partitions */
+                        if (PARTITION_IS_FOREIGN(p)) {
+                                p->new_uuid = p->current_uuid;
 
-                        if (p->current_label) {
-                                p->new_label = strdup(p->current_label);
-                                if (!p->new_label)
-                                        return log_oom();
-                        }
+                                if (p->current_label) {
+                                        p->new_label = strdup(p->current_label);
+                                        if (!p->new_label)
+                                                return log_oom();
+                                }
 
-                        continue;
-                }
+                                continue;
+                        }
 
-                if (!sd_id128_is_null(p->current_uuid))
-                        p->new_uuid = p->current_uuid; /* Never change initialized UUIDs */
-                else {
-                        r = partition_acquire_uuid(context, p, &p->new_uuid);
-                        if (r < 0)
-                                return r;
+                        if (!sd_id128_is_null(p->current_uuid))
+                                p->new_uuid = p->current_uuid; /* Never change initialized UUIDs */
+                        else {
```

Comment:
this branch should be skipped if p->new_uuid is already non-NULL

Reply:
I do not think this is necessary: We would not be in this branch if p->new_uuid was non-NULL.

Reply:
Code has changed to pretty much what you had suggested earlier.